### DISCUSSION TITLE
Error when taking the union of genes

### DISCUSSION THREAD

**Discussion by haiderabbas678** [brianhie/scanorama#154] [Q&A]
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
Cell In[213], line 4
      2 lis=[Nuc1, Nuc2, R_Ctrl, R_14B, R_14C, R_70B, R_70C, R_120A, R_120B]
      3 corrected=scanorama.integrate_scanpy(lis, union=True)
----> 4 corrected=scanorama.correct_scanpy(lis, union=True)

File [~/miniconda3/lib/python3.11/site-packages/scanorama/scanorama.py:249](http://localhost:8888/home/mirzah/miniconda3/lib/python3.11/site-packages/scanorama/scanorama.py#line=248), in correct_scanpy(adatas, **kwargs)
    245 adata.var_names = genes
    246 gene2idx = { gene: idx for idx, gene in
    247              zip(adatas[i].var.index,
    248                  adatas[i].var_names.values) }
--> 249 var_idx = [ gene2idx[gene] for gene in genes ]
    250 adata.var = adatas[i].var.loc[var_idx]
    252 adata.uns = adatas[i].uns

File [~/miniconda3/lib/python3.11/site-packages/scanorama/scanorama.py:249](http://localhost:8888/home/mirzah/miniconda3/lib/python3.11/site-packages/scanorama/scanorama.py#line=248), in <listcomp>(.0)
    245 adata.var_names = genes
    246 gene2idx = { gene: idx for idx, gene in
    247              zip(adatas[i].var.index,
    248                  adatas[i].var_names.values) }
--> 249 var_idx = [ gene2idx[gene] for gene in genes ]
    250 adata.var = adatas[i].var.loc[var_idx]
    252 adata.uns = adatas[i].uns

KeyError: '0610005C13Rik'

So I took the union of genes in my anndata sample but i got this error and do not understand what the problem is. Plz advise?
corrected=scanorama.integrate_scanpy(lis, union=True)
corrected=scanorama.correct_scanpy(lis, union=True)

**Comment by NicoHadas** [comment]
I am dealing with a similar issue in my scanpy pipeline as well when setting Union=True.

**Reply by brianhie** [reply]
This looks like a bug, I can look into it but it will probably take me a long time to get to this task. In the meantime, you might want to take the intersection. Also note that taking the union will impute zeros in datasets where the gene is not observed -- not sure if that is a correct assumption for your task.

**Reply by NicoHadas** [reply]
Hello,

Any update on this fix yet?

Thanks!
